### PR TITLE
More Map Change Related Documentaion

### PR DESCRIPTION
**Pull request by adrienntindall**
Still trying to get back into the swing of things so keeping it small, will update till I get some feedback mostly on the new names

### CODE REVIEW

**Comment by lhearachel** [comment]
Just noting that this needs to be run through clang-format; it cannot be merged until that is done

File: include/constants/field/map_load.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#ifndef POKEPLATINUM_CONSTANTS_FIELD_MAP_LOAD_H
+#define POKEPLATINUM_CONSTANTS_FIELD_MAP_LOAD_H
+
+enum {
+    MAP_LOAD_TYPE_OVERWORLD,
+    MAP_LOAD_TYPE_UNDERGROUND,
+    MAP_LOAD_TYPE_UNION,
+    MAP_LOAD_TYPE_COLOSSEUM,
+    MAP_LOAD_TYPE_BATTLE_TOWER,
+    MAP_LOAD_TYPE_MAX
```
Comment (lhearachel) [inline include/constants/field/map_load.h:10]: **question:** Will these ever be invoked in field scripts? Curious if we need to canonize these in `consts` definitions yet.

File: include/constants/field/map_load.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#ifndef POKEPLATINUM_CONSTANTS_FIELD_MAP_LOAD_H
+#define POKEPLATINUM_CONSTANTS_FIELD_MAP_LOAD_H
+
+enum {
+    MAP_LOAD_TYPE_OVERWORLD,
```
Comment (lhearachel) [inline include/constants/field/map_load.h:5]: **polish:** Canonize the first `enum` definition with `= 0`.

File: include/struct_decls/struct_02054F44_decl.h [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #ifndef POKEPLATINUM_STRUCT_02054F44_DECL_H
 #define POKEPLATINUM_STRUCT_02054F44_DECL_H
 
-typedef struct UnkStruct_02054F44_t UnkStruct_02054F44;
+typedef struct TerrainCollisionManager_t TerrainCollisionManager;
```
Comment (lhearachel) [inline include/struct_decls/struct_02054F44_decl.h:4]: **polish:** Can this be zipped into the same header as its definition?

File: include/struct_defs/struct_020EC3A8.h [hunk 4]
```diff
@@ -2,14 +2,14 @@
...
+    u32 fieldBottomScreen : 4;
     u32 unk_00_4 : 4;
     u32 unk_00_8 : 4;
     u32 unk_00_12 : 4;
     u32 unk_00_16 : 4;
     u32 unk_00_20 : 4;
     u32 unk_00_24 : 8;
     u32 unk_04;
-} UnkStruct_020EC3A8;
+} MapLoadMode;
```
Comment (lhearachel) [inline include/struct_defs/struct_020EC3A8.h:13]: **polish:** Canonize the struct name above (`struct MapLoadMode`).

File: src/field_menu.c [hunk 5]
```diff
@@ -332,9 +332,9 @@ void sub_0203AB00(FieldSystem *fieldSystem)
         menu->unk_224 = sub_0203AC28(fieldSystem);
     } else if (sub_0206C0D0(fieldSystem) == 1) {
         menu->unk_224 = sub_0203AC2C(fieldSystem);
-    } else if (fieldSystem->unk_70 == 3) {
+    } else if (fieldSystem->mapLoadType == 3) {
         menu->unk_224 = sub_0203AC3C(fieldSystem);
-    } else if (fieldSystem->unk_70 == 2) {
+    } else if (fieldSystem->mapLoadType == 2) {
```
Comment (lhearachel) [inline src/field_menu.c:337]: **polish:** Do these not use the constants you defined in this PR?

File: src/field_system.c [hunk 6]
```diff
@@ -376,8 +378,8 @@ struct PoketchSystem *FieldSystem_GetPoketchSystem(void)
 
 BGL *sub_0203D170(void *param0)
 {
-    FieldSystem *v0 = (FieldSystem *)param0;
-    return v0->unk_08;
+    FieldSystem *fieldSystem = (FieldSystem *)param0;
+    return fieldSystem->unk_08;
```
Comment (lhearachel) [inline src/field_system.c:382]: **polish:** Does this match as:

```c
return ((FieldSystem *)param0)->unk_08;
```

File: src/journal.c [hunk 7]
```diff
@@ -119,22 +119,22 @@ int Journal_SaveSize(void)
     return sizeof(Journal) * 10;
 }
 
-static void sub_0202B60C(Journal *param0)
+static void Jounral_Init(Journal *journal)
```
Comment (lhearachel) [inline src/journal.c:122]: **spelling**:

```suggestion
static void Journal_Init(Journal *journal)
```

File: src/field_map_change.c [hunk 8]
```diff
@@ -79,15 +80,15 @@
...
-} UnkStruct_02053900;
+} MapChangeData;
 
 typedef struct {
-    int unk_00;
-    Location unk_04;
-} UnkStruct_02053A80;
+    int state;
+    Location nextLocation;
+} MapChangeSubData;
```
Comment (lhearachel) [inline src/field_map_change.c:91]: **polish:** Canonize these struct names (`struct X { ...`). Applies to other instances in the file.

File: src/field_map_change.c [hunk 9]
```diff
@@ -607,30 +606,30 @@ void FieldSystem_StartLoadMapFromErrorTask(FieldSystem *fieldSystem)
...
+    Location *location = &mapChangeData->nextLocation;
 
-    switch (v1->unk_00) {
+    switch (mapChangeData->state) {
     case 0:
         Sound_PlayEffect(1539);
         Sound_TryFadeInBGM(fieldSystem, location->mapId);
         sub_02055974(taskMan);
-        (v1->unk_00)++;
+        (mapChangeData->state)++;
```
Comment (lhearachel) [inline src/field_map_change.c:628]: **polish**: Remove unnecessary parens (applies to other instances in this file).

**Review by lhearachel — CHANGES_REQUESTED** [review]

File: include/constants/field/map_load.h (hunk 1 above)
Comment (adrienntindall) [reply include/constants/field/map_load.h:10]: unsure

File: include/struct_decls/struct_02054F44_decl.h (hunk 3 above)
Comment (adrienntindall) [reply include/struct_decls/struct_02054F44_decl.h:4]: This struct is defined in a c file as of right now, I can move it to a header if that's desired

File: include/constants/field/map_load.h (hunk 1 above)
Comment (lhearachel) [reply include/constants/field/map_load.h]: Fine to leave as is for now ,then.

File: include/struct_decls/struct_02054F44_decl.h (hunk 3 above)
Comment (lhearachel) [reply include/struct_decls/struct_02054F44_decl.h:4]: I'd like for us to compress structs into headers wherever possible, and it's easiest to do that as we bump into them.

File: src/field_overworld_state.c [hunk 10]
```diff
@@ -84,7 +84,7 @@ Location *sub_0203A72C(FieldOverworldState *fieldState)
     return &fieldState->unk_50;
 }
 
-Location *sub_0203A730(FieldOverworldState *fieldState)
+Location *FieldOverworldState_GetSpecialLocation(FieldOverworldState *fieldState)
 {
     return &fieldState->unk_3C;
```
Comment (lhearachel) [inline src/field_overworld_state.c:89]: **polish**: Label this field if you can give it an accessor.

File: src/journal.c [hunk 11]
```diff
@@ -150,7 +150,7 @@ Journal *Journal_GetSavedPage(Journal *journal, BOOL param1)
             journal[v1] = journal[v1 - 1];
         }
 
-        Jounral_Init(&journal[0]);
+        Journal_Init(&journal[0]);
```
Comment (lhearachel) [inline src/journal.c:153]:
```suggestion
        Journal_Init(journal);
```

**Review by lhearachel — CHANGES_REQUESTED** [review]
Just a few nitpicks

File: src/journal.c (hunk 11 above)
Comment (adrienntindall) [reply src/journal.c:153]: if you look up in the function, journal is an array, so it makes sense to write it like this

**Review by lhearachel — APPROVED** [review]

**Pull request merged**